Improved Itemized Layer

Pedinel opened the pull request:
This changes allows ItemizedLayer to add multiple Items of any collections.

Explanation:
if I have this code (CustomMarkerItem extends MarkerItem):
`ItemizedLayer<MarkerItem> markerLayer = new ItemizedLayer<>(map, symbol);
LinkedList<CustomMarkerItem> myIconsList = new LinkedList<>();
`

I can't add my Icon list to the ItemizedLayer:
`markerLayer.addItems(myIconsList);`

andreynovikov committed:
Circle map style rendering (#202) #122

devemux86 committed:
Add location layer and Samples example #171

devemux86 committed:
Enable location layer rendering when needed #171

andreynovikov committed:
Polygon label position enhancements (#204), closes #80

devemux86 committed:
libGDX desktop: disable OpenAL

devemux86 committed:
Polygon label position (minor improvements) (#204), #80

devemux86 committed:
0.6.0-rc2

andreynovikov committed:
Polygon label position (#204), #80
- Fix infinite recursion on invalid polygons
- Allow setting area-size for text styles
- Skip unnecessary calculations if label is outside of visible area

Commit:
LocationRenderer: revert shader precision from high to medium #209, fixes #207

devemux86 committed:
Update Android and Gradle plugin versions

devemux86 committed:
0.6.0-rc3

devemux86 commented:
It's impossible to check the PR with all that history of unrelated commits.
Can you create a [topic branch](https://help.github.com/articles/about-pull-requests/) based on latest master and have only your changes in a commit?

devemux86 commented:
VTM development continues at https://github.com/mapsforge/vtm
(see announcement here on main page)

Could you post your idea there?

Pedinel1 committed:
ItemizedLayer: API enhancements for adding Collection of any MarkerInterface implementation (#211)

devemux86 committed:
0.6.0

devemux86 committed:
Update changelog

devemux86 committed:
Render theme improvements #41

devemux86 committed:
Mapbox vector tiles by @hjanetzek #57

devemux86 committed:
Mapbox vector tiles improvements and example #57

devemux86 committed:
Fix render theme schema 'src' regular expression

devemux86 committed:
Mapzem .mvt theme: update with SVG and various improvements #57

devemux86 committed:
Mapbox tile source: use new Mapzen production url #57

devemux86 committed:
MapFileTileSource: zoom level improvements, closes #219

devemux86 committed:
PathLayer: draw line improvements by @Bezzu, fixes #108

devemux86 committed:
Update changelog #108

devemux86 committed:
Mapsforge: multiple map files, closes #208

devemux86 committed:
MercatorProjection.latitudeToY not work at > ±85.05113°, fixes #222

Pedinel1 committed:
PathLayer: API enhancements for adding Collection of any GeoPoint extension (#211)

devemux86 committed:
Update documentation

devemux86 committed:
Render themes: use tessellation in sea #208

devemux86 committed:
Mapsforge: multiple map files with zoom levels, #208

devemux86 committed:
Samples: minor improvements #32

devemux86 committed:
Gradle: declare compile time only dependencies with Java plugin #229

devemux86 committed:
Update dependencies

devemux86 committed:
Fix house numbers (nodes) visibility (with new maps), fixes #168

devemux86 committed:
MapDatabase: use TileClipper buffer as half the tile size, #231

devemux86 committed:
Render theme styles sample #93

mathieudebrito committed:
Android samples: add fragment example (#233)

devemux86 committed:
Render theme styles sample improvements #93

devemux86 committed:
Update Travis

devemux86 committed:
Render themes improvements #41

devemux86 committed:
Mapbox vector tiles: use tessellation in water #57

devemux86 committed:
Render themes improvements #41

devemux86 committed:
Update changelog

devemux86 committed:
MapDatabase: clear tags with each element read, #234, fixes #235

devemux86 committed:
House numbers (nodes) mix with symbols, fixes #234

devemux86 committed:
Tile grid layer scaling, closes #238

devemux86 committed:
Use baseline 160dpi in scaling, closes #236

devemux86 committed:
Render themes improvements #41

devemux86 committed:
Tile grid enhance color #238

devemux86 committed:
MapDatabase: use TileClipper buffer based on dpi, #231

devemux86 committed:
Update Android support lib

devemux86 committed:
Render themes improvements #41

devemux86 committed:
AwtPaint: fix getColor

devemux86 committed:
Location layer: lazy rendering start for performance #171

devemux86 committed:
Osmarender theme: fix 'scrub' overlap without tessellation '#41

devemux86 committed:
Update badges

schedul-xor committed:
Update vtm-web modules (#240) #51

schedul-xor committed:
UrlTileSource: use StringBuilder instead of String concatenation (#241)

devemux86 committed:
Improve code / xml formatting #54

devemux86 committed:
vtm-web: update documentation

schedul-xor committed:
Made {X},{Y},{Z} customizable in derived tile sources for tile URL (#244)

schedul-xor committed:
Support TextureRegion to be used for MarkerSymbol (#245)

devemux86 committed:
vtm-web modules: minor cleanup #51

schedul-xor committed:
Added easing functions which currently works with Animator (#242)

devemux86 committed:
Animator easing: improvements and backwards API compatibility (#242) #246

devemux86 committed:
Fix Travis build

schedul-xor committed:
Check and clamp bearing range each time it's set (#248)

andreynovikov committed:
Improved gestures (#249) #253

devemux86 committed:
vtm-app: fix location layer

devemux86 committed:
Marker examples: add (map) long press events #32

devemux86 committed:
Android: fix quick scale vs long press, fixes #250

devemux86 committed:
Samples: map events improvements #32

devemux86 committed:
Update Android build tools

devemux86 committed:
Gradle: enhance clean tasks in vtm web apps

devemux86 committed:
Samples cleanup

devemux86 committed:
Render themes new SVG resources, closes #251

mathieudebrito committed:
Add sample showing MapPosition animations not merge (#252)

devemux86 committed:
GDX: improve keyboard shortcuts

devemux86 committed:
Improved gestures: parallel system and samples #253

devemux86 committed:
Samples: add map events in desktop marker example #32

Longri committed:
Add events for scale / rotate / tilt by user (#256)

andreynovikov committed:
Improved gestures: use VTM Task for asynchronous calls (#260) #253

andreynovikov committed:
Fix zoom animation not executed sometimes (#261)

devemux86 committed:
OSMUtils: include correct copyrights #35

andreynovikov committed:
OSMUtils: Treat amenity as area in heuristic (#262)

devemux86 committed:
Samples reorganization #32

devemux86 committed:
MapPosition animation sample improvements #252

devemux86 committed:
Android example for render theme styles #93, closes #255

devemux86 committed:
GDX double tap zoom, closes #263

devemux86 committed:
MapPosition animation sample with queue

devemux86 committed:
Map.getMapPosition at animation end option

devemux86 committed:
Circle map style #122

devemux86 committed:
GL shader version, closes #265

devemux86 committed:
Render theme improvements #41

devemux86 committed:
Update changelog

devemux86 committed:
Update Gradle Android plugin

devemux86 committed:
CircleStyle: add style builder #122

devemux86 committed:
CircleStyle fix #122

devemux86 committed:
SVG resources scaling, closes #214

devemux86 committed:
SymbolStyle: add style builder #214

devemux86 committed:
Circle map style: use GL quads or GL points #122

devemux86 committed:
LocationLayer: move lazy enable to application #171

devemux86 committed:
Applications list: update an app's license

devemux86 committed:
Update applications list

devemux86 committed:
Update readme

devemux86 committed:
Render buckets improvements

devemux86 committed:
Render styles improvements

devemux86 committed:
ExtrusionStyle: add StyleBuilder interface

devemux86 committed:
TextStyle improvements

devemux86 committed:
Update Android build tools

devemux86 committed:
Circles (GL points) anti-aliasing #268

devemux86 committed:
Update Android support library

devemux86 committed:
Circles (GL points) stroke #268

devemux86 committed:
Update libGDX to 1.9.5 #270

devemux86 committed:
Render theme callback, closes #274

devemux86 committed:
Update applications list

devemux86 committed:
Update changelog

devemux86 committed:
Oneway arrows in themes, closes #275

devemux86 committed:
Map limit via bounding box, closes #277

boldtrn committed:
Mapbox (Mapzen) vector tiles (#279), #57, fixes #271

devemux86 committed:
Mapbox (Mapzen) vector tiles improvements #57

devemux86 committed:
Render themes improvements #41

devemux86 committed:
Mapbox (Mapzen): locale option #57

devemux86 committed:
Mapbox (Mapzen): locale improvements #57

Longri committed:
iOS: improve upload texture for IosBitmap (#282)

devemux86 committed:
Add Mapsforge map providers list doc

devemux86 committed:
Update integration guide

devemux86 committed:
Update changelog

Longri committed:
Create TextureAtlas with TextureRegions from bitmaps (#283)

Longri committed:
Texture atlas with marker overlays example (#284)

devemux86 committed:
Update readme

Longri committed:
Add platform identification (#286)

devemux86 committed:
Platform identification improvements #286, closes #285

boldtrn committed:
Render capital cities early (#287)

Mapbox (Mapzen): Render capital cities early

devemux86 committed:
Mapbox (Mapzen) theme: cities improvements #287 #57

devemux86 committed:
Update Android support library

devemux86 committed:
Update integration guide

schedul-xor committed:
libGDX1.9.5, GWT2.8.0 applied to vtm-web (#290)

devemux86 committed:
Update vtm-web-js build

devemux86 committed:
Samples: skip z-order in fragment #32

devemux86 committed:
Samples: fix location layer visibility #32

devemux86 committed:
vtm-web-app: include theme with texture atlas symbols #51

devemux86 committed:
Update readme

devemux86 committed:
Fix readme

devemux86 committed:
OkHttp 3 engine, closes #138

devemux86 committed:
Mapbox (Mapzen): use OkHttp engine #57

devemux86 committed:
OkHttp: catch exceptions #138

devemux86 committed:
TileSource: add name parameter

devemux86 committed:
Android MapView: add onDestroy method

devemux86 committed:
Update Mapsforge-Maps.md

devemux86 committed:
vtm-web: add in Maven deploy

devemux86 committed:
vtm-web: fix javadoc

devemux86 committed:
0.7.0-rc1

Longri committed:
Symbol rotation (#294)

nebular committed:
Graphics API: draw circle (#295)

nebular committed:
Marker clustering (#296) (#310) #312

devemux86 committed:
XmlThemeBuilder: use English case transformation rules, fixes #297

devemux86 committed:
0.7.0-rc2

devemux86 committed:
Update JTS

devemux86 committed:
Update Android support library

devemux86 committed:
Make Travis happy

mathieudebrito committed:
Made OKHttpEngine compliant with TileCache (#299) #135

Longri committed:
Allow multiple TextureAtlas per SymbolBucket (#303)

devemux86 committed:
0.7.0

Longri committed:
Texture atlas: use bitmap packer based on libGdx PixmapPacker (#304)

schedul-xor committed:
WebGL: skip GL_VERTEX_PROGRAM_POINT_SIZE for GL points (#305)

nebular committed:
New Osmagray theme (#306), closes #300

- Based on OSMARENDER, just translated all colors to their grayscale equivalent
- Does not include textures, the few textures like parks will show up in color
- ... but 95% of the style is grayscale

devemux86 committed:
Update changelog

devemux86 committed:
Symbol rotation #294 improvements

devemux86 committed:
Use Osmagray theme in samples #300

boldtrn committed:
Mapzen theme: show neighbourhoods (#308) #57

devemux86 committed:
Mapzen theme: add regions #57

devemux86 committed:
OkHttp external cache improvements, closes #135

Longri committed:
Theme texture atlas at runtime (#324), #63

devemux86 committed:
Graphics API: draw circle Desktop implementation

devemux86 committed:
Graphics API: draw line using floats

devemux86 committed:
Update changelog

Longri committed:
iOS: canvas drawCircle implementation (#309)

devemux86 committed:
Atlas multi-texture Android example #301

devemux86 committed:
Android samples: use proper inheritance, fixes #311

devemux86 committed:
Mapsforge sample: add scale bar #32

devemux86 committed:
Canvas fillColor: use common defaults, closes #314

devemux86 committed:
Samples: use vector map #32

Longri committed:
vtm-ios-example module (#326)

devemux86 committed:
Gradle 3.3, Android Gradle plugin 2.3.0, Android Maven plugin 1.5

devemux86 committed:
Update Mapsforge maps

mathieudebrito committed:
LocationRenderer: add reverse version of shader 1 (#318)

devemux86 committed:
Update changelog

devemux86 committed:
Marker clustering improvements #312

devemux86 committed:
Graphics API: enable paint text alignment, closes #320

devemux86 committed:
Location renderer: allow external shaders, move existing to glsl files, closes #317

devemux86 committed:
Location shaders: check no rotation, fixes #321

Longri committed:
BitmapPacker wrong Texture on iOS (#323), fixes #322

devemux86 committed:
Atlas multi-texture sample improvements #319

devemux86 committed:
Location renderer improvements #321

devemux86 committed:
Theme texture atlas #324 improvements, #63

devemux86 committed:
vtm-ios-example Gradle cleanup

devemux86 committed:
Update changelog and readme

devemux86 committed:
Samples improvements #32

devemux86 committed:
Samples improvements #32

devemux86 committed:
Update readme

Longri committed:
AwtCanvas draw bitmap pixel perfect (#328)

Longri committed:
iOS: remove unused Gradle settings (#329)

devemux86 committed:
Graphics API #328 improvements

devemux86 committed:
Update integration guide

devemux86 committed:
Android MapView in layout designer, closes #330

devemux86 committed:
Update integration guide

devemux86 committed:
Location renderer improvements #321

Longri committed:
IosBitmap.recycle() crash (#332)

devemux86 committed:
Integration guide: add Android (with libGDX)

devemux86 committed:
libGDX 1.9.6, closes #333

devemux86 committed:
Update integration guide

devemux86 committed:
Update integration guide

Longri committed:
POT textures (#334)

Longri committed:
vtm-ios: remove slf4j-simple dependency (#335)

devemux86 committed:
POT textures #334 improvements and samples

devemux86 committed:
Update applications list

devemux86 committed:
Update Android support library

devemux86 committed:
Update changelog

glorantq committed:
vtm-playground: add SpriteBatch test (#337)

devemux86 committed:
GdxMap: make map renderer protected

devemux86 committed:
Gretty plugin (Gradle) at web modules, closes #338

devemux86 committed:
Android samples: fix scale bar check on exit #32

Longri committed:
Handle layer enabled state (#342)

devemux86 committed:
Layer enabled state #342 improvements

devemux86 committed:
Update Android support library

devemux86 committed:
Location renderer improvements #321

Longri committed:
Map renderer COORD_SCALE configurable and global use (#344)

devemux86 committed:
MapboxMapActivity: disable cache, add grid for debug #32

devemux86 committed:
Update applications list

devemux86 committed:
Fix applications list

devemux86 committed:
Update changelog

devemux86 committed:
Update Android Gradle plugin

devemux86 committed:
Update applications list

Longri committed:
make iOS 10.3 and Xcode 8.3 compatible (#345)

recompile SvgGh library with new code singature needed by iOS 10.3

devemux86 committed:
GdxSpriteBatchTest: fix map position scale #337

devemux86 committed:
Update Android build tools

devemux86 committed:
Improve canvas DPI setting, fixes #349

devemux86 committed:
Update applications list

Longri committed:
Update RoboVM to 2.3.1 release (#350)

devemux86 committed:
Update Android Gradle plugin

mathieudebrito committed:
Added accessible static value as default building height (#351)

devemux86 committed:
Update applications list

devemux86 committed:
Render themes: add highway ref tag #41

devemux86 committed:
Update dependencies

devemux86 committed:
Render theme styles: allow setting menu callback #93

devemux86 committed:
OSMIndoorLayer: enable labels, fixes #355

devemux86 committed:
OSMIndoorLayer: add style scaling #355

devemux86 committed:
UrlTileDataSource: catch NPE input stream

devemux86 committed:
OkHttpEngine: add timeout options #138

devemux86 committed:
Mapsforge: remove MapDatabase.readZoomTable redundant checks, fixes #357
